rotkehlchen/balances/historical.py

```diff
@@ -431,6 +444,12 @@ def _update_balances(
                 spend_amount = event.amount
                 receive_amount = event.amount * event.rate  # type: ignore[assignment]  # AssetAmount is the same as fval
 
+            if assets is not None:  # Only include amounts for the specified assets.
+                if spend_asset not in assets:
+                    spend_amount = ZERO
+                if receive_asset not in assets:
+                    receive_amount = ZERO  # type: ignore[assignment]  # AssetAmount is the same as fval
+
```

Comment:
```suggestion
           if isinstance(event, Trade):
        if event.trade_type in {TradeType.BUY, TradeType.SETTLEMENT_BUY}:
            spend_asset = event.quote_asset
            receive_asset = event.base_asset
            spend_amount = event.amount * event.rate
            receive_amount = event.amount
        else:
            spend_asset = event.base_asset
            receive_asset = event.quote_asset
            spend_amount = event.amount
            receive_amount = event.amount * event.rate  # type: ignore[assignment]  # AssetAmount is the same as fval

        # Only process amounts for tracked assets
        spend_amount = ZERO if (assets is not None and spend_asset not in assets) else spend_amount
        receive_amount = ZERO if (assets is not None and receive_asset not in assets) else receive_amount  # type: ignore[assignment]
```
formatting is a bit off in my suggestion.